src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java

```diff
@@ -1305,6 +1307,26 @@ public static String firstPage(String pages) {
     }
 
     /**
+     * Return the non-digit prefix of pages
+     *
+     * @param pages
+     *            a pages string such as L42--111 or L7,41,73--97 or L43+
+     *
+     * @return the non-digit prefix of pages (like "L" of L7)
+     *         or "" if no non-digit prefix is found in the string
+     *
+     * @throws NullPointerException
+     *             if pages is null.
+     */
+    public static String pagePrefix(String pages) {
+        if (pages.matches("^\\D+.*$")) {
```

Comment:
@Siedlerchr Thanks a lot for the comment. It would be very useful to me. I've found the same comment [here](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html).

> though for repeated matches it is less efficient since it does not allow the compiled pattern to be reused.